tests/e2e/Services/VCS/VCSConsoleClientTest.php

```diff
@@ -0,0 +1,316 @@
+<?php
+
+namespace Tests\E2E\Services\VCS;
+
+use Tests\E2E\Scopes\Scope;
+use Tests\E2E\Client;
+use Tests\E2E\Scopes\ProjectCustom;
+use Tests\E2E\Scopes\SideConsole;
+use Utopia\App;
+use Utopia\Database\Helpers\ID;
+use Utopia\Database\Helpers\Role;
+use Utopia\VCS\Adapter\Git\GitHub;
+use Utopia\Cache\Adapter\None;
+use Utopia\Cache\Cache;
+
+class VCSConsoleClientTest extends Scope
+{
+    use ProjectCustom;
+    use SideConsole;
+
+    public ?string $providerRepositoryId = null;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->providerRepositoryId = '705764267';
+    }
+
+    public function testGitHubAuthorize()
+    {
+        /**
+         * Test for SUCCESS
+         */
+        $response = $this->client->call(Client::METHOD_GET, '/mock/github/callback', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installation_id' => App::getEnv('_APP_VCS_TEST_GITHUB_INSTALLATION_ID'),
+            'projectId' => $this->getProject()['$id'],
+        ]);
+
+        $this->assertNotEmpty($response['body']['installationId']);
+        $installationId = $response['body']['installationId'];
+        return $installationId;
+    }
+
+    /**
+     * @depends testGitHubAuthorize
+     */
+    public function testGetInstallation(string $installationId)
+    {
+        /**
+         * Test for SUCCESS
+         */
+
+        $installation = $this->client->call(Client::METHOD_GET, '/vcs/installations/' . $installationId, array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => $installationId
+        ]);
+
+        $this->assertEquals(200, $installation['headers']['status-code']);
```

Comment:
Let's add extra checks. Check value of provider, project ID, personal, providerInstallationId.. See what all is in the response, and assert what is expected.

Reply:
The detect runtime only returns the detected runtime in response.